mam/mam/message.c

```diff
@@ -733,6 +735,10 @@ retcode_t mam_msg_read_header(mam_msg_read_context_t *const ctx,
                                trits_from_rep(MAM_CHANNEL_ID_SIZE, chid)),
         ret);
     ERR_GUARD_RETURN(0 == ver, RC_MAM_VERSION_NOT_SUPPORTED);
+
+    if (!mam_pk_t_set_contains(trusted_channels_pks, &chid)) {
+      return RC_MAM_PK_IS_NOT_TRUSTED;
+    }
```

Comment:
Actually, do we really need to check here ? Do we need chid to be trusted to read a regular message ? Shouldn't we verify that only when engaging in delegation of trust i.e. chid1 or epid1 ?

Reply:
you're right